### PR TITLE
Annotation & XFA: Scale the font size in choicelist using zoom factor (bug 1715996)

### PR DESCRIPTION
**Pull request by calixteman**
- this is an accessibility issue which could be painful for some people with visual disabilities.

### CODE REVIEW

File: web/viewer.css [hunk 1]
```diff
@@ -16,6 +16,8 @@
 @import url(pdf_viewer.css);
 
 :root {
+  --zoom-factor: 1;
```
Comment (Snuffleupagus) [inline web/viewer.css:19]: This needs to be placed in `pdf_viewer.css` instead, since otherwise you're breaking the standalone viewer components.

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]

File: src/core/xfa/template.js [hunk 2]
```diff
@@ -1293,6 +1293,7 @@ class ChoiceList extends XFAObject {
     const style = toStyle(this, "border", "margin");
     const ui = this[$getParent]();
     const field = ui[$getParent]();
+    const fontSize = (field.font && field.font.size) || 10;
```
[the remark this answers is not in the document]
Comment (calixteman) [reply src/core/xfa/template.js:1296]: In XFA, the default font-size is 10 (see specs for v3.3) so it's why.

**Comment by Snuffleupagus** [comment]
There seem to be one regression in `chrome-annotation-choice-widget-forms-page1`.

**Comment by calixteman** [comment]
Yep, I forgot to set the font size only for combo choicelist.

**Comment by Snuffleupagus** [comment]
> Yep, I forgot to set the font size only for combo choicelist.

It looks like you forgot to push the changes?

Also, a couple of the `integrationtest`s fail on all platforms/browser, so those tests may need some updates as well?

**Comment by Snuffleupagus** [comment]
Testing the preview, with the `annotation-choice-widget.pdf` file from the test-suite, there's now a large of amount of "padding" in the dropdowns (i.e. *similar* to the previously failing ref-tests above); please see the following screen-shot:

image

**Comment by calixteman** [comment]
> Testing the preview, with the `annotation-choice-widget.pdf` file from the test-suite, there's now a large of amount of "padding" in the dropdowns (i.e. _similar_ to the previously failing ref-tests above); please see the following screen-shot:
> 

I thought adding font-size stuff on an `optgroup` will help to reduce the number of setProperty but I misunderstood what `optgroup` is and it's why some extra blanks were added at the beginning of the line.
So finally, I added font-size stuff on each `option`, if you've a good idea to avoid that, please propose.

**Comment by calixteman** [comment]
The differences are due to the fact we use the fontsize from the pdf.

**Comment by Snuffleupagus** [comment]
> So finally, I added font-size stuff on each `option`, if you've a good idea to avoid that, please propose.

Given that documents likely won't contain more than perhaps a handful of these form elements per page, and that we unload pages that aren't visible, I'm not sure if this really is a big problem (unfortunately I don't have a better idea here).

**Comment by Snuffleupagus** [comment]
> So finally, I added font-size stuff on each `option`, if you've a good idea to avoid that, please propose.

Given that documents likely won't contain more than perhaps a handful of these form elements per page, and that we unload pages that aren't visible, I'm not sure if this really is a going to be a big problem.
Unfortunately I've not got a better idea off the top of my head, so let's do this for now (and I'll try to fix the standalone `PDFPageView` case in a follow-up).

**Pull request merged**